esp-wifi/src/lib.rs

```diff
@@ -348,15 +361,61 @@ pub fn initialize(
     }
 }
 
+pub fn deinit(
+    _init: EspWifiInitialization,
+) -> Result<(TimeBase, hal::peripherals::RADIO_CLK), InitializationError> {
+    unsafe { deinit_unsafe() }
+}
+
+pub unsafe fn deinit_unsafe() -> Result<(TimeBase, hal::peripherals::RADIO_CLK), InitializationError>
+{
+    // Disable coexistence
+    #[cfg(coex)]
+    {
+        unsafe { crate::wifi::os_adapter::coex_disable() };
+        unsafe { crate::wifi::os_adapter::coex_deinit() };
+    }
+
+    // Deinitialize WiFi
+    #[cfg(feature = "wifi")]
+    {
+        esp_wifi_result!(unsafe { esp_wifi_stop() })?;
+        esp_wifi_result!(unsafe { esp_wifi_deinit_internal() })?;
+    }
+
+    // Deinitialize BLE
+    #[cfg(feature = "ble")]
```

Comment:
Don't we have to check what was initialized before? Can you just call `ble_deinit` even if it wasn't initialized? Same goes for wifi and coex, too.

Reply:
At this point we've agreed on the principle of “shutdown everything”, as I recall? 🤔 
This is also why this function is `unsafe`. Also, at this stage of `esp-wifi` implementation I'm personally not sure how exactly we can keep track of what has been initialized. I have a couple ideas-to-try, but I think it will be done outside of this PR.

Reply:
It's fine to shut down everything, my question was whether it's okay to shut down something that wasn't turned on in the first place.